packages/builtin/string.pony

```diff
@@ -237,6 +237,13 @@ class val String is (Seq[U8] & Comparable[String box] & Stringable)
     null terminator. Does not check for null terminators inside the string.
     """
     _size = len.min(_alloc - 1)
+    let alloc = (_size + 1).next_pow2().min(len.max_value() - 1)
```

Comment:
From sync call: This code shouldn't end up getting run. We won't end up freeing memory. So the code in truncate can from what we see, be dropped.